main.js

```diff
@@ -0,0 +1,98 @@
+
+let listContainer = document.getElementById('restaurant-list')
+let restLink = document.getElementById("rest-link")
+
+/*------------------Adds the map---------------------------------------*/
+
+let myMap = L.map('map').setView([44.4780344,-73.2141866], 16);
+let OpenStreetMap_Mapnik = L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+    maxZoom: 19,
+    attribution: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
+}).addTo(myMap);
+
+/*------------------Function to get the addresses------------------------*/
+
+async function getAddress() {
+    let info = await fetch('https://json-server.burlingtoncodeacademy.now.sh/restaurants')
+        .then((data) => {
+            return data.json()
+        })
+        .then((jsonObj) => {
+            return jsonObj
+        })
+        info.forEach((restaurant) => {
+            getLatLon(restaurant.address)
+            
+        })
+
+//-------------------------------Put names on markers------------------------------------------------
+
+//        async function getName() {
```

Comment:
You were on the right track here. Consider making some sort of variable to pass into your getName function, and what that variable might do. If the markers had some sort of **container** you could point to, you could **append** variables like the address, phone #, etc. and then **bind** some sort of pop-up to the aforementioned variable at each marker's location.